server/SocketManager/socketmanager.js

```diff
@@ -1,11 +1,14 @@
 let sockets = {};
+let socket_user_map = {};
 
-const updateSocket = (usertoken, clientsocket) => {
+const updateSocket = (usertoken, clientsocket, user_id) => {
   sockets[usertoken] = clientsocket;
+  socket_user_map[usertoken] = user_id;
```

Comment:
The `updateSocket` function now accepts an additional parameter `user_id`, which is a good enhancement. However, ensure that the `user_id` is validated before being used to prevent potential issues with invalid or undefined values. Consider adding a check to handle such cases.


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*